### PR TITLE
Day 4 - Getting started with React (Component and Tools)

### PR DESCRIPTION
**Pull request by AnimeshKumarSuccessive**
Animesh Kumar
Ticket number: [40385](https://redmine.successive.tech/issues/40385)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import TextFieldDemo from './pages/TextFieldDemo/TextFieldDemo';
+
+/* eslint-disable react/react-in-jsx-scope */
```
Comment (yogeshsinghh) [inline src/App.jsx]: remove eslint disable and import React

File: src/components/TextField/TextField.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { input, error, errorBorder } from './style';
+
+const TextField = (props) => {
+  // console.log(props);
```
Comment (yogeshsinghh) [inline src/components/TextField/TextField.jsx]: remove console

File: src/components/TextField/TextField.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  const { value, errorMessage, disabled } = props;
```
Comment (yogeshsinghh) [inline src/components/TextField/TextField.jsx:6]: change errorMessage with the error

File: src/components/TextField/TextField.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <p style={error}>{errorMessage}</p>
+    </div>
+  );
+};
+
+TextField.propTypes = {
+  value: PropTypes.string.isRequired,
+  errorMessage: PropTypes.string.isRequired,
+  disabled: PropTypes.bool.isRequired,
+};
```
Comment (yogeshsinghh) [inline src/components/TextField/TextField.jsx]: these props are not required and should be default values by the propTypes

File: src/index.jsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById('root'),
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (yogeshsinghh) [inline src/index.jsx]: remove these comments

**Review by yogeshsinghh — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (AnimeshKumarSuccessive) [reply src/App.jsx]: done

File: src/components/TextField/TextField.jsx (hunk 2 above)
Comment (AnimeshKumarSuccessive) [reply src/components/TextField/TextField.jsx]: Done

File: src/components/TextField/TextField.jsx (hunk 3 above)
Comment (AnimeshKumarSuccessive) [reply src/components/TextField/TextField.jsx:6]: error is already declared in the line no. 3

File: src/index.jsx (hunk 5 above)
Comment (AnimeshKumarSuccessive) [reply src/index.jsx]: Done

File: src/components/TextField/TextField.jsx (hunk 4 above)
Comment (AnimeshKumarSuccessive) [reply src/components/TextField/TextField.jsx]: Done

**Review by yogeshsinghh — APPROVED** [review]

**Pull request merged**